search/search_tests_support/test_search_engine.hpp

```diff
@@ -29,6 +29,8 @@ class TestSearchEngine : public Index
   TestSearchEngine(string const & locale, unique_ptr<storage::CountryInfoGetter> && infoGetter);
   TestSearchEngine(string const & locale, unique_ptr<storage::CountryInfoGetter> && infoGetter,
                    unique_ptr<search::SearchQueryFactory> && factory);
+  TestSearchEngine(string const & locale, unique_ptr<::search::SearchQueryFactory> && factory);
```

Comment:
Pass unique ptr by value, not by rvalue-reference.
Also, feel free to fix other constructors in this file.